onnxruntime/core/optimizer/split_quickgelu_fusion.cc

```diff
@@ -0,0 +1,145 @@
+    return false;
+  }
+
+  // check shape information is not available for input
+  Node& split_node = node;
+  NodeArg* input = split_node.MutableInputDefs()[0];
+  const TensorShapeProto* S = input->Shape();
+  if (S == nullptr || S->dim_size() < 1) {
+    return false;
+  }
+
+  // Split supports only float/float16/double/bfloat16 (OR MORE!!!!!????) - see ./onnxruntime/core/graph/contrib_ops/contrib_defs.cc
+  auto type_allowed = [](NodeArg* input) {
+    auto data_type = input->TypeAsProto()->tensor_type().elem_type();
+    if (data_type != ONNX_NAMESPACE::TensorProto_DataType_DOUBLE &&
+        data_type != ONNX_NAMESPACE::TensorProto_DataType_FLOAT16 &&
+        data_type != ONNX_NAMESPACE::TensorProto_DataType_FLOAT &&
+        data_type != ONNX_NAMESPACE::TensorProto_DataType_BFLOAT16) {
+      return false;
+    }
+    return true;
+  };
+  if (!type_allowed(input)) {
+    return false;
+  }
+
+  // Trying to find Split->QuickGelu->Mul Path
+  std::vector<const Node::EdgeEnd*> edges;
+  std::vector<graph_utils::EdgeEndToMatch> quickgelu_mul_path{
+      {0, 0, "QuickGelu", {1, 11, 13}, kOnnxDomain},
+      {0, 0, "Mul", {7, 13, 14}, kOnnxDomain}};
+
+  if (!graph_utils::FindPath(node, true, quickgelu_mul_path, edges, logger)) {
+    DEBUG_LOG("Failed to find path for QuickGelu mul operation.");
+    return false;
+  }
+  for (size_t i = 0; i < edges.size(); i++) {
+    if (!optimizer_utils::CheckOutputEdges(graph, edges[i]->GetNode(), 1)) {
+      DEBUG_LOG("Output edge count not expected for nodes.");
+      return false;
+    }
+  }
+  Node& quickgelu_node = *graph.GetNode(edges[0]->GetNode().Index());
+  Node& mul_node = *graph.GetNode(edges[1]->GetNode().Index());
+
+  std::vector<graph_utils::EdgeEndToMatch> only_mul_path{
+      {0, 0, "Mul", {7, 13, 14}, kOnnxDomain}};
+
+  if (!graph_utils::FindPath(node, true, only_mul_path, edges, logger)) {
+    DEBUG_LOG("Failed to find for direct Mul.");
+    return false;
+  }
+  for (size_t i = 0; i < edges.size(); i++) {
+    if (!optimizer_utils::CheckOutputEdges(graph, edges[i]->GetNode(), 1)) {
+      DEBUG_LOG("Output edge count not expected for nodes.");
+      return false;
+    }
+  }
+
+  Node& mul_node_2 = *graph.GetNode(edges[0]->GetNode().Index());
+
+  // Compare if the two mul_nodes are same
+  // Figure this out?
+
+
+  // pattern match succeeded
+  split = &split_node;
+  quickgelu = &quickgelu_node;
+  mult = &mul_node;
+  return true;
+}
+
+// coalesce subgraph nodes into fused node
+void FuseSplitQuickGeluSubgraph(
+    Graph& graph,
+    Node& split_node,
+    Node& quickgelu_node,
+    Node& mul_node,
+    NodeArg* input,
+    int axis,
+    int alpha) {
+  std::string fused_desc =
+      "fused " + split_node.Name() + " and " + quickgelu_node.Name() + " and " + mul.Name() + " into SplitQuickGelu";
+
+  std::string op_type = "SplitQuickGelu";
+  Node& fused_node = graph.AddNode(graph.GenerateNodeName(op_type),
```

Comment:
nit: add transformer name into node's name to make future debugging easier to figure out which graph transformer generate the node.